lib/capistrano/tasks/bundler.cap

```diff
@@ -1,40 +1,67 @@
-        with the --deployment and --quiet flags.
+        Configure the Bundler environment for the release so that subequent
+        `bundle check`, `bundle install`, `bundle clean`, and `bundle exec`
+        commands all behave consistently. The following settings will be
+        turned into the appropriate `bundle config` executions:
+
+        :bundle_config
+        :bundle_gemfile
+        :bundle_path
+        :bundle_without
+    DESC
+  task :config do
+    on fetch(:bundle_servers) do
+      within release_path do
+        with fetch(:bundle_env_variables) do
+          configuration = fetch(:bundle_config).dup || {}
+          configuration[:gemfile] = fetch(:bundle_gemfile)
+          configuration[:path] = fetch(:bundle_path)
+          configuration[:without] = fetch(:bundle_without)
+
+          configuration.each do |key, value|
+            execute :bundle, "config", "--local", key, value.to_s.shellescape unless value.nil?
+          end
+        end
+      end
+    end
+  end
+
+  desc <<-DESC
+        Install the current Bundler environment. By default, gems will be
+        installed to the shared/bundle path. Gems in the development and
+        test group will not be installed. The install command is executed
+        with the --quiet and --jobs=4 flags.
 
         By default, bundler will not be run on servers with no_release: true.
 
         You can override any of these defaults by setting the variables shown below.
 
           set :bundle_roles, :all
 
+          set :bundle_config, { deployment: true }
           set :bundle_servers, -> { release_roles(fetch(:bundle_roles)) }
-          set :bundle_binstubs, -> { shared_path.join('bin') }
+          set :bundle_binstubs, nil
           set :bundle_gemfile, -> { release_path.join('Gemfile') }
           set :bundle_path, -> { shared_path.join('bundle') }
           set :bundle_without, %w{development test}.join(' ')
-          set :bundle_flags, '--deployment --quiet'
-          set :bundle_jobs, nil
+          set :bundle_flags, '--quiet'
+          set :bundle_jobs, 4
           set :bundle_env_variables, {}
           set :bundle_clean_options, ""
     DESC
-  task :install do
+  task install: :config do
     on fetch(:bundle_servers) do
       within release_path do
         with fetch(:bundle_env_variables) do
-          options = []
-          options << "--gemfile #{fetch(:bundle_gemfile)}" if fetch(:bundle_gemfile)
-          options << "--path #{fetch(:bundle_path)}" if fetch(:bundle_path)
-
-          if fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
+          if fetch(:bundle_check_before_install) && test(:bundle, :check)
             info "The Gemfile's dependencies are satisfied, skipping installation"
           else
+            options = []
             options << "--binstubs #{fetch(:bundle_binstubs)}" if fetch(:bundle_binstubs)
             options << "--jobs #{fetch(:bundle_jobs)}" if fetch(:bundle_jobs)
```

Comment:
How about using `nproc` here as a default instead of a number?

Reply:
Good suggestion. Bundler actually [defaults to number of processors](https://github.com/rubygems/bundler/pull/5986/files) if you don't pass `--jobs`, so maybe Capistrano's default should be `nil`?

In any case I'd like to make that change as a separate PR, since it is unrelated to this PR. Can we save this discussion for later?

Reply:
Maybe just not introduce this change then? https://github.com/capistrano/bundler/pull/122/files#diff-91990b1845e3608397d87393341f0544R51

Reply:
That was not a code change, it is just documenting the existing behavior: https://github.com/capistrano/bundler/blob/d785960d82bb41c796dffdb910eb28ca86425bfb/lib/capistrano/tasks/bundler.cap#L85

Reply:
Ah sorry my bad, yup in that case makes much more sense to do it in a different PR

Reply:
I created an issue so that I remember to come back to it: https://github.com/capistrano/bundler/issues/123